common/lib/capa/capa/util.py

```diff
@@ -195,3 +195,10 @@ def get_inner_html_from_xpath(xpath_node):
     # strips outer tag from html string
     inner_html = re.sub('(?ms)<%s[^>]*>(.*)</%s>' % (xpath_node.tag, xpath_node.tag), '\\1', html)
     return inner_html.strip()
+
+
+def remove_markup(html):
+    """
+    Return the html without markup.
+    """
+    return bleach.clean(html, tags=[], strip=True)
```

Comment:
Sort of. :)  First, it's important to note that our templates are configured to use `decode.utf8` by default, so the line I added before should make your test code work more like the platform.

Using `decode.utf8` which is a Mako filter preserves the fact that you created a `Markup` object when you called `HTML()`.  Without this, something in the template takes that `Markup` object and converts it to something like unicode (with the first escaping of `&amp;`), and it loses the fact that it was a `Markup` object, so the template will double-escape it.

I'm not sure how the `Markup` gets lost without `decode.utf8` to preserve it.  If you are really curious, you can try to load a template via the platform without that as a default, and review the python that Mako outputs in the temp directory, and see the actual code.  You can then run what it is running and you'll see where it gets lost.